### PR TITLE
Implement groupby collect_set

### PR DESCRIPTION
**Pull request by ttnghia**
This partially addresses #2973.

This PR implements groupby `collect_set` aggregation. The idea of this PR is to simply apply `drop_list_duplicates` (https://github.com/rapidsai/cudf/pull/7528) to the result generated by groupby `collect_list`, obtaining collect lists without duplicate entries.

Examples:
```
keys = {1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3};
vals = {10, 11, 10, 10, 20, 21, 21, 20, 30, 33, 32, 31};

keys_output = {1, 2, 3};
vals_output = {{10, 11}, {20, 21}, {30, 31, 32, 33}};
```

In this PR, a simple, incomplete Python binding for `collect_set` has been added, and no Java binding is implemented yet. Complete bindings for those Python/Java sides need to be implemented later in some other separate PRs.

### CODE REVIEW

**Comment by jrhemstad** [comment]
Is this only for `rolling`? If so, can you clarify in the title? Also, the PR description needs to be more complete as it is used in generating the CHANGELOG.

**Comment by ttnghia** [comment]
> Is this only for `rolling`? If so, can you clarify in the title? Also, the PR description needs to be more complete as it is used in generating the CHANGELOG.

Hi Jake. I just started to work on this thus things are still unclear to put in. I'll update more details for the description along the way.

**Comment by mythrocks** [comment]
Might I suggest a change to the title and description of this issue? Let's just tackle renaming `COLLECT` to `COLLECT_LIST`.
(This is assuming that @shwina has no objection.)

We can address `COLLECT_SET` in `groupby` and `rolling_window` as separate tasks.

**Comment by mythrocks** [comment]
Also, if this is a work in progress, it would be best to switch this to a draft PR rather than use `[WIP]` in the title.

**Comment by shwina** [comment]
Changing `COLLECT` to `COLLECT_LIST` sounds absolutely fine to me, although for now, we probably still want the name of the aggregation in Cython/Python to be `"collect"``.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

File: cpp/src/groupby/groupby.cu [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ * Copyright (c) 2019-2021, NVIDIA CORPORATION.
```
Comment (karthikeyann) [inline cpp/src/groupby/groupby.cu:2]: no changes in this file. but copyright updated.

File: cpp/src/groupby/sort/groupby.cu [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ * Copyright (c) 2019-2021, NVIDIA CORPORATION.
```
Comment (karthikeyann) [inline cpp/src/groupby/sort/groupby.cu:2]: possible merge conflict with PR #7387

**Review by karthikeyann — APPROVED** [review]
cpp code approval.

File: cpp/src/groupby/groupby.cu (hunk 1 above)
Comment (ttnghia) [reply cpp/src/groupby/groupby.cu:2]: Ah yes, I realized that this file has been changed in another recent merge.

**Comment by ttnghia** [comment]
Rerun tests.

File: cpp/src/groupby/sort/groupby.cu (hunk 2 above)
Comment (ttnghia) [reply cpp/src/groupby/sort/groupby.cu:2]: I see. Then either this, or that PR need to be merged first.

**Comment by ttnghia** [comment]
Rerun tests.

**Review by kkraus14 — APPROVED** [review]
cmake / conda lgtm, will let @shwina review the Python side 😄

**Review by jlowe — APPROVED** [review]
Java approval

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by ttnghia** [comment]
Rerun tests.

**Review by shwina — APPROVED** [review]
Cython LGTM!

**Comment by kkraus14** [comment]
@gpucibot merge

**Comment by ttnghia** [comment]
Thanks all 😃

**Pull request merged**